src/datastation/scripts/fix_provenance_files.py

```diff
@@ -28,11 +30,14 @@ def validate(xml_path: str, xsd_path: str) -> bool:
     xmlschema = etree.XMLSchema(xmlschema_doc)
     try:
         xml_doc = etree.parse(xml_path)
-        result = xmlschema.validate(xml_doc)
-        return result
-    except etree.XMLSyntaxError:
+        xmlschema.assertValid(xml_doc)
```

Comment:
this still parses the xsd on every row of the inputfile, right?

Reply:
Yes, if `validate() == False` is equivalent with `assertValid()` throwing `DocumentInvalid`.